src/models/basemodel.h

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BASEMODEL_H
+#define BASEMODEL_H
+
+#include <QByteArray>
+
+class BaseModel {
+ public:
+  BaseModel() = default;
+  virtual ~BaseModel() = default;
+
+  bool initialized() const { return !m_rawJson.isEmpty(); }
+
+  void forget() {
+    m_rawJson.clear();
+    m_writeNeeded = false;
```

Comment:
This seems like calling `forget()` could potentially leave us without the ability to erase settings from disk. If `m_writeNeeded` is false then we might never actually touch the disk to clear out what was there before.